src/components/catalog-parameters/catalog-parameters.controller.ts

```diff
@@ -74,6 +81,84 @@ export class CatalogParametersController implements angular.IController {
     }
   }
 
+  private setupFormDependencies() {
+    _.each(_.get(this.ctrl.parameterSchema, 'dependencies'), (dependency: any, key: string) => {
+      if (_.isArray(dependency.oneOf)) {
+        // Only support 'oneOf' at this time, not 'allOf' or 'anyOf'
+        // See https://github.com/mozilla-services/react-jsonschema-form#dynamic for example 'dependencies' object
+        _.each(dependency.oneOf, (schema) => {
+          this.mapDependencyCondition(schema, key);
+        });
+      } else {
+        // simple nested dependency
+        this.mapDependencyCondition(dependency, key);
+      }
+    });
+  }
+
+  private mapDependencyCondition(schema, key) {
+    // Get the fields to show if the dependency is valid
+    let requiredFields = _.get(schema, 'required');
+    if (!_.isArray(requiredFields)) {
+      return;
+    }
+    _.each(requiredFields, (requiredField: any) => {
+      let fieldIndex = _.findIndex(this.ctrl.parameterForm, (field: any) => {
+        if (_.isObject(field)) {
+          return requiredFields.indexOf(field.key) > -1;
+        } else {
+          return requiredFields.indexOf(field) > -1;
+        }
+      });
+
+      // Only continue if the field is in the form defintion
+      if (fieldIndex > -1) {
+        // Normalise the field object
+        let field = this.ctrl.parameterForm[fieldIndex];
+        if (_.isString(field)) {
+          field = {
+            key: field
+          };
+        }
+
+        // Check if the required field is already in the schema properties.
+        // Add it if not.
+        if (_.isUndefined(this.ctrl.parameterSchema.properties[field.key])) {
+          this.ctrl.parameterSchema.properties[field.key] = schema.properties[field.key];
+        }
+
+        // Set the angular 'condition'
+        let enumMatches = _.get(schema, 'properties.' + key + '.enum');
+        if (_.isUndefined(enumMatches)) {
```

Comment:
Should this just be...?

```js
if (enumMatches) {
```

Reply:
This is an undefined check, so do you mean `if (!enumMatches) {` ?